examples/megacli.cpp

```diff
@@ -3907,6 +3942,26 @@ void exec_sync(autocomplete::ACState& s)
         }
     }
 }
+
+void exec_syncconfig(autocomplete::ACState& s)
+{
+    if (s.words.size() == 1)
+    {
+        cout << "Current sync config: " << syncConfig.syncType <<
+                " " << syncConfig.syncDeletions << " " << syncConfig.forceOverwrite << endl;
+    }
+    else if (s.words.size() == 4)
+    {
+        syncConfig.syncType = static_cast<SyncConfig::Type>(atoi(s.words[1].s.c_str()));
+        syncConfig.syncDeletions = atoi(s.words[2].s.c_str());
+        syncConfig.forceOverwrite = atoi(s.words[3].s.c_str());
+        cout << "Successfully applied new sync config!" << endl;
+    }
+    else
+    {
+        assert(false);
```

Comment:
An assert here is too critical, IMHO. User might mistype the config, ie. add an extra param by mistake, and we can simply remind the user that's not a valid configuration. In other words, it's an error for the user, not for the developer that requires to halt execution.
In any case, it looks like MEGAcli handles that at the autocomplete side, no calling the this method if the provided params doesn't match the configuration of the command.

Reply:
Yea. The argument parsing already takes care of ensuring we end up with the right number of parameters. So if the assert is hit then it must be a programming error, not a user error.

Reply:
Probably, we doesn't need the assertion in that case :stuck_out_tongue: 
But alright, let's keep it